donut/modules/uploads/routes.py

```diff
@@ -0,0 +1,109 @@
+import flask
+import os
+from werkzeug import secure_filename
+from donut.modules.uploads import blueprint, helpers
+
+
+@blueprint.route('/lib/<path:url>')
+def display(url):
+    '''
+    Displays the webpages that have been created by users.
+    '''
+    page = helpers.read_page(url.replace(' ', '_'))
+    if page == None:
+        return flask.abort(404)
+    return flask.render_template(
+        'page.html',
+        title=url.replace('_', ' '),
+        permissions=helpers.check_upload_permission())
+
+
+@blueprint.route('/uploads/_send_page', methods=['GET'])
+def get_page():
+    '''
+    Sends the page to frontend.
+    '''
+    url = flask.request.args.get('url')
+    page = helpers.read_page(url.replace(' ', '_'))
+    return page
```

Comment:
No need for `page` variable

Reply:
This sends the content of the markdown file to the front end.

Reply:
woops misunderstood you